app/src/main/java/com/github/multimatum_team/multimatum/adaptater/GroupMemberAdapter.kt

```diff
@@ -0,0 +1,117 @@
+package com.github.multimatum_team.multimatum.adaptater
+
+import android.app.AlertDialog
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.github.multimatum_team.multimatum.R
+import com.github.multimatum_team.multimatum.model.UserGroup
+import com.github.multimatum_team.multimatum.model.UserInfo
+import com.github.multimatum_team.multimatum.repository.UserRepository
+import com.github.multimatum_team.multimatum.viewmodel.AuthViewModel
+import com.github.multimatum_team.multimatum.viewmodel.GroupViewModel
+import kotlinx.coroutines.runBlocking
+
+/**
+ * A RecyclerView adapter to bridge the group member list and the group view model.
+ */
+class GroupMemberAdapter(
+    private val context: Context,
+    private val userRepository: UserRepository,
+    private val authViewModel: AuthViewModel,
+    private val groupViewModel: GroupViewModel
+) : RecyclerView.Adapter<GroupMemberAdapter.MemberItemViewHolder>() {
+    private var dataSource: MutableList<UserInfo> = mutableListOf()
+    private lateinit var group: UserGroup
+
+    /**
+     * Update the group associated to the adapter.
+     */
+    fun setGroup(group: UserGroup) {
+        this.group = group
+        dataSource =
+            runBlocking { userRepository.fetch(group.members.toList()) }
+                .toMutableList()
+        notifyDataSetChanged()
+    }
+
+    /**
+     * Create view holding the user information, using the right layout.
+     */
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MemberItemViewHolder {
+        val view = LayoutInflater
+            .from(parent.context)
+            .inflate(R.layout.list_item_group_member, parent, false)
+        return MemberItemViewHolder(view)
+    }
+
+    /**
+     * Prompt the group owner for confirmation when pressing a member name.
+     */
+    private fun showConfirmMemberRemovalDialog(memberInfo: UserInfo, position: Int) {
```

Comment:
This function is not covered with the tests, is it possible to do a test quickly during this sprint ?